plugin/src/main/java/org/opensearch/ml/action/controller/UndeployModelControllerTransportAction.java

```diff
@@ -0,0 +1,121 @@
+        TransportService transportService,
+        ActionFilters actionFilters,
+        MLModelManager mlModelManager,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        Client client,
+        DiscoveryNodeHelper nodeFilter,
+        MLStats mlStats,
+        NamedXContentRegistry xContentRegistry,
+        ModelAccessControlHelper modelAccessControlHelper
+    ) {
+        super(
+            MLUndeployModelControllerAction.NAME,
+            threadPool,
+            clusterService,
+            transportService,
+            actionFilters,
+            MLUndeployModelControllerNodesRequest::new,
+            MLUndeployModelControllerNodeRequest::new,
+            ThreadPool.Names.MANAGEMENT,
+            MLUndeployModelControllerNodeResponse.class
+        );
+        this.mlModelManager = mlModelManager;
+        this.clusterService = clusterService;
+        this.client = client;
+        this.nodeFilter = nodeFilter;
+        this.mlStats = mlStats;
+        this.xContentRegistry = xContentRegistry;
+        this.modelAccessControlHelper = modelAccessControlHelper;
+    }
+
+    @Override
+    protected MLUndeployModelControllerNodesResponse newResponse(
+        MLUndeployModelControllerNodesRequest request,
+        List<MLUndeployModelControllerNodeResponse> responses,
+        List<FailedNodeException> failures
+    ) {
+        return new MLUndeployModelControllerNodesResponse(clusterService.getClusterName(), responses, failures);
+    }
+
+    @Override
+    protected MLUndeployModelControllerNodeRequest newNodeRequest(MLUndeployModelControllerNodesRequest request) {
+        return new MLUndeployModelControllerNodeRequest(request);
+    }
+
+    @Override
+    protected MLUndeployModelControllerNodeResponse newNodeResponse(StreamInput in) throws IOException {
+        return new MLUndeployModelControllerNodeResponse(in);
+    }
+
+    @Override
+    protected MLUndeployModelControllerNodeResponse nodeOperation(MLUndeployModelControllerNodeRequest request) {
+        return createUndeployModelControllerNodeResponse(request.getUndeployModelControllerNodesRequest());
+    }
+
+    private MLUndeployModelControllerNodeResponse createUndeployModelControllerNodeResponse(
+        MLUndeployModelControllerNodesRequest undeployModelControllerNodesRequest
+    ) {
+        String modelId = undeployModelControllerNodesRequest.getModelId();
+
+        Map<String, String> modelControllerUndeployStatus = new HashMap<>();
+        modelControllerUndeployStatus.put(modelId, "received");
+
+        String localNodeId = clusterService.localNode().getId();
+
+        mlModelManager.undeployModelController(modelId, ActionListener.wrap(r -> {
+            log.info("Successfully undeployed model controller for model {} on node {}", modelId, localNodeId);
+        }, e -> { log.error("Failed to undeploy model controller for model {} on node {}", modelId, localNodeId, e); }));
```

Comment:
From this line, if exception happens on node, just log it, won't throw exception. For such case, will the node added bo `undeployModelControllerNodesResponse.failures` ?

Refer to plugin/src/main/java/org/opensearch/ml/action/controller/DeleteModelControllerTransportAction.java line 234
`for (FailedNodeException failedNodeException : undeployModelControllerNodesResponse.failures()) {`

Reply:
I checked other transport on node action class, it seems they don't throw exception, only find a way to handle them, so I think it will be fine